File: 05 Matemática/1028 Figurinhas/1028.py
def MDC(num1, num2):
    if (num2 == 0):
        return num1
    else:
        return MDC(num2, num1 % num2)


cases = int(input())

for i in range(0, cases):
    num_ricardo, num_vicente = [int(x) for x in input().split(' ')]
    print(MDC(num_ricardo, num_vicente))

# https://www.beecrowd.com.br/judge/pt/problems/view/1028
# Ranking 01696º, Time 0.389, Submiss Date 06/03/2023 14:27:46 (Python 3.9)

File: 05 Matemática/1028 Figurinhas/1028.cpp
#include <bits/stdc++.h>

using namespace std;

int MDC(int num1, int num2) {
    if (num2 == 0) {
        return num1;
    } else {
        return MDC(num2, num1 % num2);
    }
}

int main() {
    int cases;
    cin >> cases;

    for (int i = 0; i < cases; ++i) {
        int num_ricardo, num_vicente;
        cin >> num_ricardo >> num_vicente;
        cout << MDC(num_ricardo, num_vicente) << endl;
    }

    return 0;
}
